drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -48,74 +49,56 @@ local function find_default_endpoint(device, component)
   table.sort(eps)
   for _, v in ipairs(eps) do
     if v ~= 0 then --0 is the matter RootNode endpoint
-      res = v
-      break
+      return v
     end
   end
   device.log.warn(string.format("Did not find default endpoint, will use endpoint %d instead", device.MATTER_DEFAULT_ENDPOINT))
   return res
 end
 
-local function initialize_switch(device)
+local function initialize_switch(driver, device)
   local switch_eps = device:get_endpoints(clusters.OnOff.ID)
   table.sort(switch_eps)
-  local component_map = {}
+  local main_endpoint = find_default_endpoint(device)
 
-  -- For switch devices, the profile components follow the naming convention "switch%d",
-  -- with the exception of "main" being the first component. Each component will then map
-  -- to the next lowest endpoint that hasn't been mapped yet.
+  local current_component_number = 2
   for i, ep in ipairs(switch_eps) do
-    if i == 1 then
-      component_map["main"] = ep
-    else
-      component_map[string.format("switch%d", i)] = ep
+    if ep ~= main_endpoint then -- don't create a child device that maps to the main endpoint
+      local name = string.format("%s %d", device.label, current_component_number)
+      device.log.info(string.format("creating child device %s", name))
+      driver:try_create_device(
+        {
+          type = "EDGE_CHILD",
+          label = name,
+          profile = "switch-binary",
+          parent_device_id = device.id,
+          parent_assigned_child_key = string.format("%02X", ep),
```

Comment:
Is there a utility function available that, provided an ep, returns its formatted value, as It is being used throughout?